    Merge pull request #90 from Sylvain303/master
    
    saving $IFS in run() not altered for code using it

diff --git a/test/bats.bats b/test/bats.bats
index 0df3810..f1aff29 100755
--- a/test/bats.bats
+++ b/test/bats.bats
@@ -30,13 +30,13 @@ fixtures bats
 @test "empty test file runs zero tests" {
   run bats "$FIXTURE_ROOT/empty.bats"
   [ $status -eq 0 ]
-  [ $output = "1..0" ]
+  [ "$output" = "1..0" ]
 }
 
 @test "one passing test" {
   run bats "$FIXTURE_ROOT/passing.bats"
   [ $status -eq 0 ]
-  [ ${lines[0]} = "1..1" ]
+  [ "${lines[0]}" = "1..1" ]
   [ "${lines[1]}" = "ok 1 a passing test" ]
 }
 
diff --git a/test/suite.bats b/test/suite.bats
index 48f7593..5371686 100755
--- a/test/suite.bats
+++ b/test/suite.bats
@@ -6,30 +6,30 @@ fixtures suite
 @test "running a suite with no test files" {
   run bats "$FIXTURE_ROOT/empty"
   [ $status -eq 0 ]
-  [ $output = "1..0" ]
+  [ "$output" = "1..0" ]
 }
 
 @test "running a suite with one test file" {
   run bats "$FIXTURE_ROOT/single"
   [ $status -eq 0 ]
-  [ ${lines[0]} = "1..1" ]
+  [ "${lines[0]}" = "1..1" ]
   [ "${lines[1]}" = "ok 1 a passing test" ]
 }
 
 @test "counting tests in a suite" {
   run bats -c "$FIXTURE_ROOT/single"
   [ $status -eq 0 ]
-  [ $output -eq 1 ]
+  [ "$output" -eq 1 ]
 
   run bats -c "$FIXTURE_ROOT/multiple"
   [ $status -eq 0 ]
-  [ $output -eq 3 ]
+  [ "$output" -eq 3 ]
 }
 
 @test "aggregated output of multiple tests in a suite" {
   run bats "$FIXTURE_ROOT/multiple"
   [ $status -eq 0 ]
-  [ ${lines[0]} = "1..3" ]
+  [ "${lines[0]}" = "1..3" ]
   echo "$output" | grep "^ok . truth"
   echo "$output" | grep "^ok . more truth"
   echo "$output" | grep "^ok . quasi-truth"
@@ -38,14 +38,14 @@ fixtures suite
 @test "a failing test in a suite results in an error exit code" {
   FLUNK=1 run bats "$FIXTURE_ROOT/multiple"
   [ $status -eq 1 ]
-  [ ${lines[0]} = "1..3" ]
+  [ "${lines[0]}" = "1..3" ]
   echo "$output" | grep "^not ok . quasi-truth"
 }
 
 @test "running an ad-hoc suite by specifying multiple test files" {
   run bats "$FIXTURE_ROOT/multiple/a.bats" "$FIXTURE_ROOT/multiple/b.bats"
   [ $status -eq 0 ]
-  [ ${lines[0]} = "1..3" ]
+  [ "${lines[0]}" = "1..3" ]
   echo "$output" | grep "^ok . truth"
   echo "$output" | grep "^ok . more truth"
   echo "$output" | grep "^ok . quasi-truth"
